kurl_proxy/cmd/main.go

```diff
@@ -206,10 +206,20 @@ func watchSecret(certs chan cert, name string, secrets corev1.SecretInterface) {
 }
 
 func getFingerprint(certData []byte) (string, error) {
-	derBlock, _ := pem.Decode(certData)
-	if derBlock == nil {
-		return "", errors.New("no PEM data found in certificate")
+	var derBlock *pem.Block
+	for {
+		derBlock, certData = pem.Decode(certData)
+		if derBlock == nil {
+			return "", errors.New("no PEM data found in certificate")
+		}
+		if derBlock.Type == "CERTIFICATE" {
+			break
+		}
+		if len(certData) == 0 {
+			return "", errors.New("no PEM data found in certificate")
```

Comment:
nit pick: this wouldn't mean there is no PEM data found, but just none of type `"CERTIFICATE"`? might be useful to change this error message so we know which code path failed. it seems unlikely that it would ever reach this case since we check for a valid cert/key pair before calling `getFingerprint`, but just a thought